core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -917,11 +916,11 @@ class GroupMetadataManager(brokerId: Int,
   }
 
   /**
-   * Gets the partition count of the group metadata topic from ZooKeeper.
-   * If the topic does not exist, the configured partition count is returned.
+   * Gets the partition count of the group metadata topic from the active controller.
+   * The topic will be created if it doesn't exist yet based on the broker settings.
    */
   private def getGroupMetadataTopicPartitionCount: Int = {
-    zkClient.getTopicPartitionCount(Topic.GROUP_METADATA_TOPIC_NAME).getOrElse(config.offsetsTopicNumPartitions)
+    controllerChannel.getPartitionCount(Topic.GROUP_METADATA_TOPIC_NAME).get
```

Comment:
This one needs to be more robust, now it's prone to timeout related errors. Working on this.